src/components/blueprint-editor-subnav/BlueprintEditorSubnav.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FC } from "react";
+import { Badge, LineTabs, EBadgePosition } from "..";
+import { IconArrowDo, IconArrowUndo, IconEdit, IconGearSettings, IconPlusCircle } from "../..";
+import { FlatActionButton } from "../flat-action-button/FlatActionButton";
+import { SBlueprintEditorSubnav } from "./styles/SBlueprintEditorSubnav";
+import { SBlueprintEditorSubnavAddButton } from "./styles/SBlueprintEditorSubnavAddButton";
+import { SBlueprintEditorSubnavConfig } from "./styles/SBlueprintEditorSubnavConfig";
+import { SBlueprintEditorSubnavConfigButtons } from "./styles/SBlueprintEditorSubnavConfigButtons";
+import { SBlueprintEditorSubnavConfigDivider } from "./styles/SBlueprintEditorSubnavConfigDivider";
+import { SBlueprintEditorSubnavTab } from "./styles/SBlueprintEditorSubnavTab";
+import { SBlueprintEditorSubnavTabs } from "./styles/SBlueprintEditorSubnavTabs";
+
+interface IBlueprintEditorSubnavGroup {
+  name: string;
+}
+
+interface IBlueprintEditorSubnav {
+  groups: IBlueprintEditorSubnavGroup[];
+  onSelectGroup: (group: string) => void;
+  selectedGroup: string;
+  onGroupAdd: () => void;
+  onClickConfig?: () => void;
+  configBadgeValue?: string;
+  onRedo?: () => void;
+  onUndo?: () => void;
+}
+
+export const BlueprintEditorSubnav: FC<IBlueprintEditorSubnav> = ({
+  groups,
+  onSelectGroup,
+  selectedGroup,
+  onGroupAdd,
+  onUndo,
+  onRedo,
+  onClickConfig,
+  configBadgeValue,
+}) => {
+  return (
+    <SBlueprintEditorSubnav>
+      <SBlueprintEditorSubnavAddButton onClick={onGroupAdd}>
+        <IconPlusCircle size={16} />
+      </SBlueprintEditorSubnavAddButton>
+
+      <SBlueprintEditorSubnavTabs>
+        {groups.map((group) => (
+          <SBlueprintEditorSubnavTab>
+            <LineTabs activated={selectedGroup === group.name} onClick={() => onSelectGroup(group.name)}>
+              {group.name}
+              <IconEdit />
```

Comment:
@nicolasshiken we may need to check if this should trigger an event to the upper component to react to this edit.